File: src/exemplos/07_Complexidade/00_Busca/01-sequencial.py
#  -*- coding: utf-8 -*-
##   @package: 01-sequencial.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Exemplos de busca sequencial em vetor.


import apc_busca as apc


lista = [1, 2, 2, 3, 5, 7, 8, 9, 9, 10]


print(('lista: {}'.format(lista)))

apc.testa_busca(-1, lista, apc.busca_sequencial_ord)
apc.testa_busca(0, lista, apc.busca_sequencial_ord)
apc.testa_busca(1, lista, apc.busca_sequencial_ord)
apc.testa_busca(2, lista, apc.busca_sequencial_ord)
apc.testa_busca(5, lista, apc.busca_sequencial_ord)
apc.testa_busca(9, lista, apc.busca_sequencial_ord)
apc.testa_busca(10, lista, apc.busca_sequencial_ord)
apc.testa_busca(11, lista, apc.busca_sequencial_ord)

print(('\nComparações acumuladas: {}\n'.format(apc.COMPARACOES_NA_BUSCA)))

File: src/exemplos/07_Complexidade/00_Busca/01-sequencial.c
/**      @file: 01-sequencial.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Exemplos de busca sequencial em vetor. */

#include <stdio.h>

#include "apc_busca.h"
#include "../../5_EstruturasDeDados/4_Vetor/apc_vetor.h"

int main() {
    int vetor[] = {1,2,2,3,5,7,8,9,9,10};

    printf("Vetor: ");
    mostra_i(vetor, 10);

    testa_busca(-1, vetor, 10, busca_sequencial_ord);
    testa_busca(0, vetor, 10, busca_sequencial_ord);
    testa_busca(1, vetor, 10, busca_sequencial_ord);
    testa_busca(2, vetor, 10, busca_sequencial_ord);
    testa_busca(5, vetor, 10, busca_sequencial_ord);
    testa_busca(9, vetor, 10, busca_sequencial_ord);
    testa_busca(10, vetor, 10, busca_sequencial_ord);
    testa_busca(11, vetor, 10, busca_sequencial_ord);

    printf("\nAcumulado: %d\n", comparacoes_na_busca);

    return 0;
}